doc/wg/core/notes/core-notes-2025-01-10.md

```diff
@@ -0,0 +1,82 @@
+- Ben: These tests are now automated for all PRs that are merged. 
+- Ben: Tyler has written a suite of tests for networking that we will be integrating into treadmill.
+- Ben: My current focus has been to improve the treadmill cli tool. I am trying to simplify the process of enqueuing jobs and selecting the hardware a given job should run on.
+- Ben: Other pain points are the login process and obtaining ssh keys etc.
+- Ben: In summary, our two aims are to allow CI testing and also provide an environment for people to remotely develop on hardware.
+- Leon: For the release tests, I added some board definitions. I copied Ben's work to also include imix and an FPGA board. This mostly worked out of the box which was great.
+- Ben: Here are some pointers to the board class implementation:
+  - https://github.com/tock/tock-hardware-ci/blob/main/hwci/boards/nrf52dk.py
+  - https://github.com/tock/tock-hardware-ci/blob/main/hwci/boards/tockloader_board.py
+- Amit: One question, say we wanted to do a release tomorrow, what would that look like using Treadmill?
+- Ben: Because Treadmill is running as PRs enter the merge queue, this will be more incremental. For a release, we would want to have a "run all tests" option that would exercise all subsystems, boards, and tests.
+- Amit: Treadmill is going to do a bunch of testing. What is missing between that and what we would want to do manually for a release?
+- Leon: Currently, we do not have all tests and boards supported, so this would require some manual testing. As we develop this further, hopefully the number of boards requiring manual testing would decrease.
+- Leon: As we add more boards and tests, I don't think we can or should run the full suite of all tests for every PR as this would take too long.
+- Leon: We need to develop an intelligent way for determining how to select which tests need to be executed for a given PR.
+- Amit: In terms of coverage, how many boards is treadmill testing?
+- Leon: We have a wip imix and litex boards. NRF is the only automated board.
+- Leon: Many of these tests are UART and GPIO. 
+- Leon: Other tests require uncommenting lines in the kernel. We haven't tackled this problem.
+- Branden: One of the original goals was parallelism. If we had multiple NRF boards or boards at different universities, what is our status with this?
+- Leon: We can add this using github actions to have a python script that uses some plan / config to match tests to available board resources.
+- Branden: Having this would help with the testing bottle neck and also improve the stability of the system.
+- Tyler: Would Treadmill's design cause headaches when deploying across universities (such as with campus IT security concerns).
+- Leon: All outgoing traffic goes through a VPN. One way we can mitigate concerns from campus IT security is by only allowing Tock CI testing jobs to run. This would mean we are not giving users access to execute arbitrary code 
```

Comment:
```suggestion
- Leon: All outgoing traffic goes through a VPN. One way we can mitigate concerns from campus IT security is by only allowing Tock CI testing jobs to run. This would mean we are not giving users direct access to execute arbitrary code 
```

This is a bit subtle, our CI workflows are still just arbitrary code being executed (hence all of our efforts to have Treadmill provide ephemeral environment where it's hard to persist changes across jobs)